### ISSUE TITLE
Finish GDB support for PRU

### ISSUE THREAD

**Issue by dinuxbg** [dinuxbg/gnupru#14]
Explore GDB PRU port feasibility. So far there is a very rough GDB backend for the PRU simulator target.

The following items remain to be implemented:
   - Stack unwiding support. This is needed for source-level GDB debugging.
   - Real target backend support via remoteproc framework, preferably without an intermediate gdbserver.

**Comment by EphDoering** [comment]
What's the status of this? I saw some news that GDB 9 has PRU emulator support, but haven't found much else about it.

**Comment by dinuxbg** [comment]
Yes, GNU Simulator has been ported to PRU. I use it to run the GCC regression testsuite on the PRU backend. You can also use it for simulation - see [this](https://github.com/dinuxbg/pru-gcc-examples/tree/master/sim-hello) example. 

FYI, GNU simulator supports tracing:
```
$ pru-run --trace --trace-insn --trace-disasm --trace-core out/pru-core.elf
...
read-4 exec:0x200004f4 -> 0x0120e2e2
memory:   read of insn 0x0120e2e2 from 200004f4
disasm:   add	sp, sp, 32
...
```

I [tried](https://github.com/dinuxbg/gnupru/blob/6ec05b2047e3ebe8b23e78ba93065ccb2a750b36/patches/binutils-gdb/0004-gdb-pru-Initial-dummy-port.patch) to port GDB but never got the time to finish it. Feel free to take over.
